src/looper.ts

```diff
@@ -35,6 +35,8 @@ const MILLISECONDS = 1000
 export const DEFAULT_THRESHOLD = 5
 const DEFAULT_REPORT_INTERVAL = 180000 // 3 minutes
 
+let probeCounter = 0
```

Comment:
I think this implementation is not right. If I assumed correctly, you extracted the `counter` variable from `loopProbe` function, right? But the problem is, that local `counter` variable is actually corresponded to each of the probes. 

For example, say we have `repeat` flag set to `1`, and we have 2 probes with different intervals, 10 seconds and 20 seconds, respectively. So Monika will actually terminate after 30 seconds, which is after both probes have run once.

So if you make the `counter` variable global like this, only the first probe will run with `repeat` flag set to `1`. This breaks the `repeat` flag feature.

Reply:
oh i see...
so the `loopCheckSTUNServer` better has it's own counter then

Reply:
fixed. please recheck